providers/sftp/src/airflow/providers/sftp/hooks/sftp.py

```diff
@@ -155,6 +177,18 @@ def get_conn_count(self) -> int:
         """Get the number of open connections."""
         return self._conn_count
 
+    def _describe_directory(self, conn: SFTPClient, path: str) -> dict[str, dict[str, str | int | None]]:
+        flist = sorted(conn.listdir_attr(path), key=lambda x: x.filename)
+        files = {}
+        for f in flist:
+            modify = datetime.datetime.fromtimestamp(f.st_mtime).strftime("%Y%m%d%H%M%S")  # type: ignore
+            files[f.filename] = {
+                "size": f.st_size,
+                "type": "dir" if stat.S_ISDIR(f.st_mode) else "file",  # type: ignore
+                "modify": modify,
+            }
+        return files
```

Comment:
```suggestion 
        return {
            f.filename: {
                "size": f.st_size,
                "type": "dir" if stat.S_ISDIR(f.st_mode) else "file",  # type: ignore
                "modify": datetime.datetime.fromtimestamp(f.st_mtime).strftime("%Y%m%d%H%M%S"),
            }
            for f in sorted(conn.listdir_attr(path), key=lambda x: x.filename)    
        }     
```

also we could make the return value a TypedDict

Reply:
Tried it and it said that TypedDict cannot be used there

Reply:
```python
class FileMetadata(TypedDict):
    file_size: str
    file_type: Literal["dir", "file"]
    modified_timestamp: str

return { 
    f.filename: FileMetadata(file_size=..., file_type=..., modified_timestamp=...)
}
```

I'm surprised the code sample above isn't working. Can you check if it works for you?

Reply:
seemed to work now, probably did something wrong beforehand